### PR TITLE
Multi key join

### PR DESCRIPTION
**Pull request by vogievetsky**

### CODE REVIEW

File: src/datatypes/dataset.ts [hunk 1]
```diff
@@ -1025,16 +1026,28 @@ export class Dataset implements Instance<DatasetValue, DatasetJS> {
...
+
+  public getKeyValueForDatum(datum: Datum): string {
+    const { keys } = this;
+    if (!keys) throw new Error('join lhs must have a key (be a product of a split)');
+    return this.keys.map(k => {
+      let v: any = datum[k];
+      if (v && v.start) v = v.start;
+      if (v && v.toISOString) v = v.toISOString();
+      return v;
+    }).join('|');
```
Comment (wylieallen-i) [inline src/datatypes/dataset.ts:1041]: Is there any particular significance to using pipe as the delimiter or could it just as well be anything?

File: src/datatypes/dataset.ts [hunk 2]
```diff
@@ -1067,46 +1077,31 @@ export class Dataset implements Instance<DatasetValue, DatasetJS> {
...
-        if (cmp < 0) {
-          newData.push(nextDatum);
-          i++;
-        } else if (cmp > 0) {
-          newData.push(nextOtherDatum);
-          j++;
+    const myDatumLookup = this.getKeyLookup();
+    const otherDatumLookup = other.getKeyLookup();
+
+    const newData = deduplicateSort(Object.keys(myDatumLookup).concat(Object.keys(otherDatumLookup))).map(keyValue => {
```
Comment (wylieallen-i) [inline src/datatypes/dataset.ts:1092]: Why call this `keyValue` instead of just `key`?

**Comment by wylieallen-i** [comment]
Should there be a test here or in imply-ui?

File: src/datatypes/dataset.ts (hunk 1 above)
Comment (vogievetsky) [reply src/datatypes/dataset.ts:1041]: nope, no significance

File: src/datatypes/dataset.ts (hunk 2 above)
Comment (vogievetsky) [reply src/datatypes/dataset.ts:1092]: you are right, it should just be key

**Pull request merged**